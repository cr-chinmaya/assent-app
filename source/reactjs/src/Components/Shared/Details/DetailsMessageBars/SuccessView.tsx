import * as React from 'react';
import * as Styled from './DetailsMessageBarsStyling';
import { Stack } from '@fluentui/react/lib/Stack';

const SuccessView = (): JSX.Element => {
    return (
        <Stack verticalAlign="center" horizontalAlign="center">
            <Stack.Item>
                <Styled.SuccessIcon iconName="SkypeCircleCheck" />
            </Stack.Item>
            <Stack.Item>
                <div
                    ref={input => input && input.focus()}
                    role="status"
                    aria-label="Your action has successfully completed"
                    title="Your action has successfully completed"
                    tabIndex={0}
                    style={{ outline: 'none' }}
                >
                    <Styled.SuccessMessage>Your action has successfully completed!</Styled.SuccessMessage>
                </div>
            </Stack.Item>
        </Stack>
    );
};

export default SuccessView;
