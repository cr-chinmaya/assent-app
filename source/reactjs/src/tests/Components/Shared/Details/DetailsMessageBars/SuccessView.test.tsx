import * as React from 'react';
import { render } from '@testing-library/react';
import SuccessView from '../../../../../Components/Shared/Details/DetailsMessageBars/SuccessView';

describe('SuccessView', () => {
    it('should render Success view correctly', () => {
        const wrapper = render(<SuccessView />);
        expect(wrapper).toMatchSnapshot();
    });
});
