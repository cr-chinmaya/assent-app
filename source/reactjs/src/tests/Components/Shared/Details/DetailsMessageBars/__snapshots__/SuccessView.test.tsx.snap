// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessView should render Success view correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-Stack css-84"
      >
        <div
          class="ms-StackItem css-85"
        >
          <i
            aria-hidden="true"
            class="ms-Icon root-33 sc-bdfBwQ gLKZdP"
            data-icon-name="SkypeCircleCheck"
            role="presentation"
          />
        </div>
        <div
          class="ms-StackItem css-85"
        >
          <div
            aria-label="Your action has successfully completed"
            role="status"
            style="outline: none;"
            tabindex="0"
            title="Your action has successfully completed"
          >
            <p
              class="sc-gsTCUz jwmOKE"
            >
              Your action has successfully completed!
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-Stack css-84"
    >
      <div
        class="ms-StackItem css-85"
      >
        <i
          aria-hidden="true"
          class="ms-Icon root-33 sc-bdfBwQ gLKZdP"
          data-icon-name="SkypeCircleCheck"
          role="presentation"
        />
      </div>
      <div
        class="ms-StackItem css-85"
      >
        <div
          aria-label="Your action has successfully completed"
          role="status"
          style="outline: none;"
          tabindex="0"
          title="Your action has successfully completed"
        >
          <p
            class="sc-gsTCUz jwmOKE"
          >
            Your action has successfully completed!
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
